Oxlint Beta (#314)
oxc-project/oxc-project.github.io: src/blog/2025-03-15-oxlint-beta.md

---
title: Oxlint Beta
outline: deep
authors:
  - boshen
  - camchenry
---

<AppBlogPostHeader />

We are thrilled to announce that oxlint is now in beta release, after more than a year of development by the community!

This milestone signifies feature completeness and stability.

## How to Use

At this stage, oxlint can be used to fully replace eslint in small to medium projects.

For larger projects, our advice is to turn off eslint rules via the eslint plugin [eslint-plugin-oxlint](https://www.npmjs.com/package/eslint-plugin-oxlint), and run oxlint before ESLint in your lint-staged or CI setup.

> To test oxlint in your JavaScript / TypeScript codebase, execute the following command at the root directory of your repository:

Reviewer: ```suggestion
To test oxlint in your codebase, execute the following command at the root directory of your repository:
```
NIT: For me this sounds like it excludes JSX, Vue, ...

::: code-group

```sh [npm]
$ npx oxlint@latest
```

```sh [pnpm]
$ pnpm dlx oxlint@latest
```

```sh [yarn]
$ yarn dlx oxlint@latest
```

```sh [bun]
$ bunx oxlint@latest
```

```sh [deno]
$ deno run npm:oxlint@latest
```

:::

Alternatively, refer to the [installation guide](/docs/guide/usage/linter) for detailed instructions.

> ## New Features since General Availability

Reviewer: Love it! 🫶 
Do we want to tell about IDE Plugins too? 
We have `vscode`, `zed`, `coc.nvim` and `intellij` (only available in preview channel).

Reviewer 2: I mentioned this as an area of future improvement, as I don't think we've truly invested a lot of time into the editor experience yet. But I think it's worth mentioning for sure.

* No performance regression.
* Over 500+ rules to choose from.
* Completed most rules from `typescript`, `unicorn`, `react`, `react-perf`, `nextjs`, `import`, `jsdoc`, `jsx-a11y`, `node`, `promise`, `jest`, `vitest` plugins.
* `.oxlintrc.json` [configuration file](/docs/guide/usage/linter/config.html#configuration-file-format).
* Lint `<script>` content in `.astro`, `.svelte` and `.vue` files.

> * [automatic fixes](/docs/guide/usage/linter/automatic-fixes.html)

Reviewer: ```suggestion
* [automatic fixes](/docs/guide/usage/linter/automatic-fixes.html).
```

its the only pullet point without a ending dot

> * [Nested config file discovery](/docs/guide/usage/linter/config.html#nesting-configuration-files).

Reviewer 3: We could make the point about perf more positive.

* Remove 1st bullet "No performance regression."
* Add this to end of the list instead: "* Despite all the new lint rules and features, oxlint is now even faster."

or alternatively "* Improved performance (even with all the new lint rules and features)".

## Next Major Release

> Custom linter plugins written in JavaScript - we have completed all the prerequisites for authoring linter plugins.

Reviewer 3: I don't think it's fair to say we have "completed" the prerequisites. There's some kinks to work out yet, and we don't have a clear timeline - so I think better to be a little vaguer here.

```suggestion
Custom linter plugins written in JavaScript. We have been making great progress laying the groundwork to support JS plugins, while maintaining good performance. Watch this space...
```

## Acknowledgements

Oxlint beta would not have been possible without the 200 contributors.

Special thank you goes to:

- [@cam314](https://github.com/camc314), [@mysteryven](https://github.com/mysteryven), [shulaoda](https://github.com/shulaoda) for implementing many sophisticated lint rules, testing, and constantly improving everything.
- [@Sysix](https://github.com/Sysix) for maintaining `eslint-plugin-oxlint`.

> - [@DonIsaac](https://github.com/DonIsaac) for improving configuration, documentation and website. For representing oxc in [Squiggle Conf](https://2024.squiggleconf.com).

Reviewer 3: Nit:

```suggestion
- [@DonIsaac](https://github.com/DonIsaac) for improving configuration, documentation and website, and for representing Oxc at [Squiggle Conf](https://2024.squiggleconf.com).
```
